scripts/core/popper-utils.tsx

```diff
@@ -0,0 +1,68 @@
+import {Modifier, Instance as PopperInstance} from '@popperjs/core';
+
+export const sameWidth: Modifier<any, any> = {
+    name: 'sameWidth',
+    enabled: true,
+    phase: 'beforeWrite',
+    requires: ['computeStyles'],
+    fn: ({state}) => {
+        state.styles.popper.width = `${state.rects.reference.width}px`;
+    },
+    effect: ({state}) => {
+        const {reference} = state.elements;
+
+        if (reference instanceof HTMLElement) {
+            state.elements.popper.style.width = `${reference.offsetWidth}px`;
+        }
+    },
+};
+
+export const applyMaxSize: Modifier<any, any> = {
+    name: 'applyMaxSize',
+    enabled: true,
+    phase: 'beforeWrite',
+    requires: ['maxSize'],
+    fn: ({state}) => {
+        const {height} = state.modifiersData.maxSize;
+
+        // subtracting 10 in order to make a gap between the edge of the viewport
+        state.styles.popper.maxHeight = `${height - 10}px`;
```

Comment:
What if someone is using a different than the standard screen size? Would `10` still be enough?

Reply:
Why would screen size matter? As you see from the comment, the goal is to make a gap - i.e. to avoid popover appearing "glued" to the bottom/top. The purpose is to make it clear visually where is the end of the popover so user doesn't try scrolling around to find the end.